firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -412,10 +412,10 @@ class BrowserViewController: UIViewController,
         } else {
             urlBar.topTabsIsShowing = showTopTabs
             urlBar.setShowToolbar(!showNavToolbar)
-            toolbar.addNewTabButton.isHidden = showNavToolbar
 
             if showNavToolbar {
                 toolbar.isHidden = false
+                toolbar.addNewTabButton.isHidden = true
```

Comment:
We indeed don't need to see it anymore. That was causing the issue because when we show the toolbar, we hide the `addNewTabButton`, which is a toolbar component. For landscape mode, we use an instance of `URLBarView`. An even better solution is to not add it to the `contentView` stack in `TabToolbar` at all.